yb-voyager/cmd/importSchemaYugabyteDB.go

```diff
@@ -90,6 +113,20 @@ func executeSqlFile(file string, objType string, skipFn func(string, string) boo
 	return nil
 }
 
+func shouldSkipDDL(stmt string) (bool, error) {
+	skipReplicaIdentity := strings.Contains(stmt, "ALTER TABLE") && strings.Contains(stmt, "REPLICA IDENTITY")
+	isNotValid, err := isNotValidConstraint(stmt)
+	if err != nil {
+		return false, fmt.Errorf("error checking whether stmt is to add not valid constraint: %v", err)
+	}
+	if (isNotValid && !bool(flagPostSnapshotImport)) || // Skipping NOT VALID CONSTRAINT in import schema without post-snapshot-mode
+		(bool(flagPostSnapshotImport) && !isNotValid) || // Skipping other TABLE DDLs in post-snapshot-import mode
+		skipReplicaIdentity { // Skip ALTER REPLICA IDENTITY always
+		return true, nil
+	}
```

Comment:
lets define boolean vars here for readability
```
skipNotValidConstraint := (isNotValid && !bool(flagPostSnapshotImport))
// similarly for other

if skipNotValidConstraint || ... {
        return true, nil...
}


```

Reply:
I also think the scenario based skip logic should go to the skipFunc() passed here.

In this function executeSql() - we were simply either using skipFn() to check  the sql, or outright ignoring if SQL is of specific syntax like `SET` or `REPLICA IDENTITY`

Reply:
That SkipFn is currently only for skipping DDLs in single execution to later, because we run this function twice once with executeSQL(skipFn) and another for executeSQL(!skipFn) 
but for these REPLICA IDENTIY we want to ignore them all the time